### PR TITLE
[FE] Fix failing tests

### PR DESCRIPTION
**Pull request by amurKontur**
## Summary by CodeRabbit

- **New Features**
	- Enhanced user registration process with additional fields for phone code and country code.
	- Introduced a new data repository for country-specific phone number formats.

- **Bug Fixes**
	- Updated locator for cookie verification to improve interaction on the Privacy Page.

- **Tests**
	- Expanded registration tests to include multiple country-specific phone formats, increasing test coverage. 

- **Documentation**
	- Added a new JSON file mapping country codes to phone number formats for testing purposes.

### CODE REVIEW

File: e2e/registration.spec.ts [hunk 1]
```diff
@@ -1,124 +1,139 @@
...
+            countryCode,
+            phone,
+            linkedin,
+          },
+        ),
+        pageManager.atKeycloakPage.getAdminToken({
+          project,
+          apiContext,
+          adminName: process.env.ADMIN_KEYCLOAK!,
+          adminPassword: process.env.ADMIN_KEYCLOAK_PASSWORD!,
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/registration.spec.ts:69]: we have this check at global setup step

File: e2e/registration.spec.ts [hunk 2]
```diff
@@ -1,124 +1,139 @@
...
-          email,
-          password,
-          phone,
-          linkedin,
-        },
-      ),
-      pageManager.atKeycloakPage.getAdminToken({
+for (const [countryCode, fullPhone] of testedPhoneByCountry) {
+  const [phoneCode, ...numbers] = fullPhone.split(' ');
+  const phone = numbers.join('');
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/registration.spec.ts:21]: we have a test data list

File: e2e/page-objects/privacyPage.ts [hunk 3]
```diff
@@ -72,9 +72,7 @@ export class PrivacyPage extends HelperBase {
       this.page.waitForURL(/cookies/),
       this.page.waitForLoadState('domcontentloaded'),
     ]);
-    await this.page
-      .locator('#cookie-files-policy-and-operational-data')
-      .scrollIntoViewIfNeeded();
+    await this.page.locator('#hdr-1').scrollIntoViewIfNeeded();
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/privacyPage.ts:75]: we'll fix it after sorting is pushed to prod

File: e2e/testsData/phone-by-country.json [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  "uy": "+598 94 231 234",
+  "uz": "+998 91 234 56 78",
+  "vu": "+678 591 2345",
+  "va": "+39 312 345 6789",
+  "ve": "+58 412 1234567",
+  "vn": "+84 912 345 678",
+  "ye": "+967 712 345 678",
+  "zm": "+260 95 5123456",
+  "zw": "+263 71 234 5678"
+}
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/testsData/phone-by-country.json:212]: I need my own data to test feature. library has it wrong

File: e2e/registration.spec.ts [hunk 5]
```diff
@@ -1,124 +1,139 @@
...
+const countriesToTest = ['us', 'ca', 'pl'];
+
 const projects = getProjects().filter((project: Project) => project.env !== 'prod');
+const phoneByCountry: [string, string][] = Object.entries(
+  getTestData('phone-by-country'),
+);
+
+const testedPhoneByCountry = phoneByCountry.filter((countryPhone) =>
+  countriesToTest.includes(countryPhone[0]),
+);
```
Comment (Natallia-Harshunova) [inline e2e/registration.spec.ts]: You can just use chaining

```
const testedPhoneByCountry = Object.entries(getTestData('phone-by-country'))
  .filter((countryPhone) => countriesToTest.includes(countryPhone[0]));
```

**Review by Natallia-Harshunova — APPROVED** [review]

File: e2e/page-objects/keycloakPage.ts [hunk 6]
```diff
@@ -76,17 +78,38 @@ export class KeycloakPage extends HelperBase {
...
-    await page.locator('#phone').fill(phone);
     await page.locator('#linkedin').fill(linkedin);
-    if (project.env !== 'dev') await page.locator('#termsAccepted').check();
-    await page.locator('[type = "submit"]').click({ force: true });
+    await page.locator('#termsAccepted').check();
+    await page.locator('#newsletter-consent').check();
+    await page.getByLabel('Selected country').click();
+    await page.getByLabel('Search').fill(phoneCode);
+    await page.locator(`[data-country-code = "${countryCode}"]`).click();
+    await page.locator('#phone').fill(phone);
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/keycloakPage.ts:101]: i need to do it just once i don't need functions

File: e2e/registration.spec.ts (hunk 5 above)
Comment (amurKontur) [reply e2e/registration.spec.ts]: firstly i had just phoneByCountry
so it's just about fast coding 

chaining could optimize HEAP storage for objects being stored. huge phoneByCountry will not be stored then

ok

Comment (amurKontur) [reply e2e/registration.spec.ts]: fixed

**Pull request merged**